src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,93 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final int RESIZE_FACTOR = 2;
+    private int size;
+    private int threshold;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = (Node<K, V>[]) new Node[DEFAULT_INITIAL_CAPACITY];
+        threshold = (int) (DEFAULT_INITIAL_CAPACITY * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int tableIndex = hash(key) % table.length;
+        Node<K, V> newNode = new Node<>(tableIndex, key, value, null);
+        if (table[tableIndex] != null) {
+            Node<K, V> temp = table[tableIndex];
+            while (temp != null) {
+                if (Objects.equals(temp.key, key)) {
+                    temp.value = value;
+                    return;
+                }
+                if (temp.next == null) {
+                    temp.next = newNode;
+                    break;
+                }
+                temp = temp.next;
+            }
+        } else {
+            table[tableIndex] = newNode;
+        }
+        if (++size > threshold) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int tableIndex = hash(key) % table.length;
```

Comment:
You have duplication of index identifying operation, method hash already makes % table.length, so no reason to do it again